scheduler.go

```diff
@@ -270,7 +270,15 @@ func (s *Scheduler) run(job *Job) error {
 		}
 		key := getFunctionKey(job.jobFunc)
 
-		locker.Lock(key)
+		success, err := locker.Lock(key)
+		if err != nil {
+			return err
+		}
+		if !success {
+			job.lastRun = s.time.Now(s.loc)
```

Comment:
Should we update lastRun here? The job didn't actually run here

Reply:
Hmmm ya. I am also quite concerned about the line when I put it. The idea was because the unsuccessful job by locking will be regarded as a skipped job, I thought the line let the job be re-scheduled for the next interval. Please let me know the approach is not good. :)